### PR TITLE
cleanup config loading method

### PR DESCRIPTION
**Pull request by hackmod**
*ready to merge*

cleanup config loading method

- use `require()` instead of `fs.readFileSync()` + `JSON.parse()`
- rename `config.json` to `config.example.json` to make it consistent with the `README.md` (e97bef88c58bfc79385cc33e8c8253a0c341f1ae)
- cleaup

### CODE REVIEW

**Comment by realcodywburns** [comment]
lgtm

File: tools/tokens/daoTxns.js [hunk 1]
```diff
@@ -17,10 +17,29 @@ var DAOCreatedToken = mongoose.model('DAOCreatedToken');
 var DAOTransferToken = mongoose.model('DAOTransferToken');
 var InternalTx     = mongoose.model( 'InternalTransaction' );
 
+// load config.json
+var config = { nodeAddr: 'localhost', gethPort: 8545 };
```
Comment (YazzyYaz) [inline tools/tokens/daoTxns.js]: Thanks for your code contribution. I will approve it, but wondering if you can refactor it first. It seems to me that it's the same piece of code added to all those different files. A more optimal way to do it is to have on file with that piece of logic that you import into the required files needed.

So, say you have this file here:
```
var config = { nodeAddr: 'localhost', gethPort: 8545 };
try {
    var local = require('../../config.json');
    _.extend(config, local);
    console.log('config.json found.');
} catch (error) {
    if (error.code === 'MODULE_NOT_FOUND') {
        var local = require('../../config.example.json');
        _.extend(config, local);
        console.log('No config file found. Using default configuration... (config.example.json)');
    } else {
        throw error;
        process.exit(1);
    }
}
```
All you need to do is just import the `config` from that separate file into all the other files. It's also more beneficial as you don't need to have different references to your `config.json` path like here: `var local = require('../../config.example.json');`

Let me know if you have any questions. It's the more engineering approach, trying to abstract everything whenever you can. It's less code to read that way and becomes a more optimal code base. I'll be making comments over where I think you can abstract the code.

File: tools/sync.js [hunk 2]
```diff
@@ -287,38 +287,28 @@ var checkBlockDBExistsThenWrite = function(config, patchData, flush) {
...
 **/
-var config = {};
-//Look for config.json file if not
+/**
+ * nodeAddr: node address
+ * gethPort: geth port
+ * bulkSize: size of array in block to use bulk operation
+ */
+// load config.json
+var config = { nodeAddr: 'localhost', gethPort: 8545, bulkSize: 100 };
```
Comment (YazzyYaz) [inline tools/sync.js]: Can abstract here to a separate file you can import.

File: tools/stats.js [hunk 3]
```diff
@@ -146,6 +142,27 @@ if (process.env.RESCAN) {
     rescan = true;
 }
 
+// load config.json
+var config = { nodeAddr: 'localhost', gethPort: 8545, bulkSize: 100 };
```
Comment (YazzyYaz) [inline tools/stats.js]: Can also abstract here to a separate file you can import.

File: tools/patcher.js [hunk 4]
```diff
@@ -151,32 +146,24 @@ var blockIter = function(web3, firstBlock, lastBlock, config) {
     }
 }
 
-var config = {};
-
+// load config.json
+var config = { nodeAddr: 'localhost', gethPort: 8545 };
```
Comment (YazzyYaz) [inline tools/patcher.js]: Can also abstract here to a separate file you can import.

File: tools/blockHelper.js [hunk 5]
```diff
@@ -179,31 +174,26 @@ var writeTransactionsToDB = function(config, blockData) {
     }
 }
 
-var config = {};
-
+// load config.json
+var config = { nodeAddr: 'localhost', gethPort: 8545, bulkSize: 100 };
```
Comment (YazzyYaz) [inline tools/blockHelper.js]: Can also abstract here to a separate file you can import.

File: routes/web3relay.js [hunk 6]
```diff
@@ -17,33 +16,25 @@ var filterBlocks = require('./filters').filterBlocks;
 var filterTrace = require('./filters').filterTrace;
 
 /*Start config for node connection and sync*/
-var config = {};
-//Look for config.json file if not
+// load config.json
+var config = { nodeAddr: 'localhost', gethPort: 8545 };
```
Comment (YazzyYaz) [inline routes/web3relay.js]: Can also abstract here to a separate file you can import.

**Comment by YazzyYaz** [comment]
@hackmod let me know if you have any questions. Basically, you can have the code chunk as a function in a separate file, that you can call from your other files. Calling the function executes your code chunk and then returns `config` which you can then use in the other files.

**Comment by hackmod** [comment]
@YazzyYaz // thank you for your kind review!

as you can see, this PR changes the `fs.readFileSync('config.json')` method to `require('config.json')`.
(minimal fix to minimize any possible breaks on forks.)
and I guess, @realcodywburns have some related improvements on his repo.

**Comment by YazzyYaz** [comment]
@hackmod makes sense, I can see you just modified the code minimally and didn't actually contribute the code chunks. I'll approve it on Monday after I test it.

**Pull request merged**